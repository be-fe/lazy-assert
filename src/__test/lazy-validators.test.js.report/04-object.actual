{}
  01 {} : true : {}
    result : true
  02 {} : false : {}
    key : "a"
    message : "Target failed to match the object validator"
    result : false
    subResult : {}
      message : "Target value is not of the expected type"
      result : false
      target : undefined
      type : "number"
    target : undefined
  03 {} : true : {}
    result : true
  04 {} : true : {}
    result : true
  05 {a: 1} : true, number : {}
    result : true
  06 {a: 1} : true, object | number : {}
    result : true
  07 {a: 1} : true, number & func : {}
    result : true
  08 {a: 1} : false, object : {}
    key : "a"
    message : "Target failed to match the object validator"
    result : false
    subResult : {}
      message : "Did not match or-array"
      result : false
      target : 1
      validator : []
        0 : "object"
    target : 1
  09 {a: 1} : false, number & func : {}
    key : "a"
    message : "Target failed to match the object validator"
    result : false
    subResult : {}
      index : 2
      message : "Did not meet all validators in and array"
      result : false
      subResult : {}
        funcInfo : undefined
        result : false
        target : 1
      subValidator : "_[[[function]]]_"
    target : 1
  10 {a: 1} : false, number & < : {}
    key : "a"
    message : "Target failed to match the object validator"
    result : false
    subResult : {}
      index : 2
      message : "Did not meet all validators in and array"
      result : false
      subResult : {}
        funcInfo : "Comparison : < 0"
        result : false
        target : 1
      subValidator : []
        0 : "<"
        1 : 0
    target : 1
  11 {a: 1} : true, number & > : {}
    result : true
  12 {a: 1} : false, object & > : {}
    key : "a"
    message : "Target failed to match the object validator"
    result : false
    subResult : {}
      index : 1
      message : "Did not meet all validators in and array"
      result : false
      subResult : {}
        message : "Target value is not of the expected type"
        result : false
        target : 1
        type : "object"
      subValidator : "object"
    target : 1
  13 {a: 1} : true, object | > : {}
    result : true
  14 {a: {b: 1}} : true : {}
    result : true
  15 {a: {b: 1}} : true : {}
    result : true
  16 {a: {b: 1}} : true : {}
    result : true
  17 {a: {b: 1}} : false : {}
    key : "a"
    message : "Target failed to match the object validator"
    result : false
    subResult : {}
      key : "b"
      message : "Target failed to match the object validator"
      result : false
      subResult : {}
        index : 1
        message : "Did not meet all validators in and array"
        result : false
        subResult : {}
          message : "Target value is not of the expected type"
          result : false
          target : 1
          type : "object"
        subValidator : "object"
      target : 1
    target : {}
      b : 1
  18 {a: {b: ["1", "2", null]]}} : false : {}
    key : "a"
    message : "Target failed to match the object validator"
    result : false
    subResult : {}
      key : "b"
      message : "Target failed to match the object validator"
      result : false
      subResult : {}
        message : "Did not match or-array"
        result : false
        target : []
          0 : "1"
          1 : "2"
          2 : null
        validator : []
          0 : "string"
          1 : []
            0 : "array"
            1 : "string"
      target : []
        0 : "1"
        1 : "2"
        2 : null
    target : {}
      b : []
        0 : "1"
        1 : "2"
        2 : null
  19 {a: {b: ["1", "2", null]]}} : true : {}
    result : true
  20 {a: {b: "1"}} : true : {}
    result : true