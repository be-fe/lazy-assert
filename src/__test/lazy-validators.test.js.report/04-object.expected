{}
  01 {} : true : {}
    result : true
  02 {} : false : {}
    key : "a"
    message : "Failed on object property validation"
    reason : "failed-on-key"
    result : false
    subResult : []
      0 : {}
        message : "Target value is not of the expected type"
        reason : "number"
        result : false
        type : "type"
    type : "object"
  03 {} : true : {}
    result : true
  04 {} : true : {}
    result : true
  05 {a: 1} : true, number : {}
    result : true
  06 {a: 1} : true, object | number : {}
    result : true
  07 {a: 1} : true, number & func : {}
    result : true
  08 {a: 1} : false, object : {}
    key : "a"
    message : "Failed on object property validation"
    reason : "failed-on-key"
    result : false
    subResult : []
      0 : {}
        message : "Did not match or-array"
        reason : "all-types-failed"
        result : false
        subResult : []
          0 : {}
            message : "Target value is not of the expected type"
            reason : "object"
            result : false
            type : "type"
            vIndex : 0
        type : "or-array"
    type : "object"
  09 {a: 1} : false, number & func : {}
    key : "a"
    message : "Failed on object property validation"
    reason : "failed-on-key"
    result : false
    subResult : []
      0 : {}
        message : "Did not meet all validators in and array"
        reason : "some-failed"
        result : false
        subResult : []
          0 : {}
            funcInfo : undefined
            message : "Target did not pass the function check"
            reason : "returned-false"
            result : false
            type : "function"
            vIndex : 2
        type : "and-array"
    type : "object"
  10 {a: 1} : false, number & < : {}
    key : "a"
    message : "Failed on object property validation"
    reason : "failed-on-key"
    result : false
    subResult : []
      0 : {}
        message : "Did not meet all validators in and array"
        reason : "some-failed"
        result : false
        subResult : []
          0 : {}
            funcInfo : "Comparison : < 0"
            message : "Target did not pass the function check"
            reason : "returned-false"
            result : false
            type : "function"
            vIndex : 2
        type : "and-array"
    type : "object"
  11 {a: 1} : true, number & > : {}
    result : true
  12 {a: 1} : false, object & > : {}
    key : "a"
    message : "Failed on object property validation"
    reason : "failed-on-key"
    result : false
    subResult : []
      0 : {}
        message : "Did not meet all validators in and array"
        reason : "some-failed"
        result : false
        subResult : []
          0 : {}
            message : "Target value is not of the expected type"
            reason : "object"
            result : false
            type : "type"
            vIndex : 1
        type : "and-array"
    type : "object"
  13 {a: 1} : true, object | > : {}
    result : true
  14 {a: {b: 1}} : true : {}
    result : true
  15 {a: {b: 1}} : true : {}
    result : true
  16 {a: {b: 1}} : true : {}
    result : true
  17 {a: {b: 1}} : false : {}
    key : "a"
    message : "Failed on object property validation"
    reason : "failed-on-key"
    result : false
    subResult : []
      0 : {}
        key : "b"
        message : "Failed on object property validation"
        reason : "failed-on-key"
        result : false
        subResult : []
          0 : {}
            message : "Did not meet all validators in and array"
            reason : "some-failed"
            result : false
            subResult : []
              0 : {}
                message : "Target value is not of the expected type"
                reason : "object"
                result : false
                type : "type"
                vIndex : 1
            type : "and-array"
        type : "object"
    type : "object"
  18 {a: {b: ["1", "2", null]]}} : false : {}
    key : "a"
    message : "Failed on object property validation"
    reason : "failed-on-key"
    result : false
    subResult : []
      0 : {}
        key : "b"
        message : "Failed on object property validation"
        reason : "failed-on-key"
        result : false
        subResult : []
          0 : {}
            message : "Did not match or-array"
            reason : "all-types-failed"
            result : false
            subResult : []
              0 : {}
                message : "Target value is not of the expected type"
                reason : "string"
                result : false
                type : "type"
                vIndex : 0
              1 : {}
                index : 2
                message : "Target does not meet the array validators"
                reason : "all-type-failed"
                result : false
                subResult : []
                  0 : {}
                    message : "Target value is not of the expected type"
                    reason : "string"
                    result : false
                    type : "type"
                    vIndex : 1
                type : "array-array"
                vIndex : 1
            type : "or-array"
        type : "object"
    type : "object"
  19 {a: {b: ["1", "2", null]]}} : true : {}
    result : true
  20 {a: {b: "1"}} : true : {}
    result : true